src/rust/ide/view/graph-editor/src/component/visualization/container/action_bar.rs

```diff
@@ -70,6 +73,144 @@ mod background {
+            let arrow_head        = arrow_head.translate_y(arrow_head_offset);
+
+            let arrow_heads = &arrow_head;
+            let arrow_heads = arrow_heads + arrow_head.rotate(2.0  * (PI/4.0).radians());
+            let arrow_heads = arrow_heads + arrow_head.rotate(4.0  * (PI/4.0).radians());
+            let arrow_heads = arrow_heads + arrow_head.rotate(6.0  * (PI/4.0).radians());
+
+            let color_path = theme::graph_editor::visualization::action_bar::icon;
+            let fill_color = style.get_color(color_path);
+            let icon       = (arrow_heads + cross).fill(color::Rgba::from(fill_color));
+
+            (background + icon).into()
+        }
+    }
+}
+
+
+mod reset_position_icon {
+    use super::*;
+
+    use std::f32::consts::PI;
+    const PIN_THORN_WIDTH: f32 = 1.0;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+            let width              = Var::<Pixels>::from("input_size.x");
+            let height             = Var::<Pixels>::from("input_size.y");
+            let background         = Rect((&width,&height)).fill(HOVER_COLOR);
+
+            let pin_head_size  = &height / 3.0;
+            let pin_head_base  = Rect((&pin_head_size,&pin_head_size));
+            let pin_head_top   = Triangle(&pin_head_size * 1.5,&pin_head_size / 2.0) ;
+            let pin_head_top   = pin_head_top.translate_y(-&pin_head_size/2.0);
+            let pin_head       = (pin_head_base + pin_head_top).translate_y(&pin_head_size/2.0);
+
+            let pin_thorn_size  = &height / 3.0;
+            let pin_thorn_width = PIN_THORN_WIDTH.px();
+            let pin_thorn       = Triangle(pin_thorn_width,pin_thorn_size);
+            let pin_thorn       = pin_thorn.rotate((PI).radians());
+            let pin_thorn       = pin_thorn.translate_y(-&pin_head_size/2.0);
+
+            let color_path = theme::graph_editor::visualization::action_bar::icon;
+            let fill_color = style.get_color(color_path);
+            let icon       = (pin_thorn + pin_head).fill(color::Rgba::from(fill_color));
+            let icon       = icon.rotate((PI/4.0).radians());
+            let icon       = icon.fill(color::Rgba::from(fill_color));
+
+            (background + icon).into()
+        }
+    }
+}
+
+#[derive(Clone,CloneRef,Debug)]
+struct Icons {
+    display_object      : display::object::Instance,
+    icon_root           : display::object::Instance,
+    reset_position_icon : ShapeView<reset_position_icon::Shape>,
+    drag_icon           : ShapeView<drag_icon::Shape>,
+    size                : Rc<Cell<Vector2>>,
+}
+
+impl Icons {
+    fn new(logger:impl AnyLogger, app:&Application) -> Self {
+        let scene               = app.display.scene();
+        let logger              = Logger::sub(logger,"Icons");
+        let display_object      = display::object::Instance::new(&logger);
+        let icon_root           = display::object::Instance::new(&logger);
```

Comment:
what do you need `icon_root` if you already have `display_object` here?

Reply:
The `display_object` is exposed externally and used to set the position of the container. Having the `icon_root` gives us a simple way to have a zero point in that parent coordinate system and avoids us having to remember the initial parent coordinates when dragging, greatly simplifying the logic here. 

Example: the container, i.e., the `display_object` is set to the centre of the parent node at `150,0`. Now we start dragging to `200,0`. If we want to reset without the `icon_root` we now need to know that we need to revert the `display_object` to `150,0`, with the `icon_root` we just reset the `icon_root` to `0,0`.

Reply:
Oh, of course, it makes sense. Would you be so nice and add short docs to it?